src/app/pages/press/press-excerpt/press-excerpt.js

```diff
@@ -1,42 +1,41 @@
-import componentType from '~/helpers/controller/init-mixin';
-import Byline from '~/components/byline/byline';
-import {description as template} from './press-excerpt.html';
-import css from './press-excerpt.css';
+import React, {useRef, useEffect} from 'react';
+import {BylineJsx} from '~/components/byline/byline';
+import './press-excerpt.css';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
 
-const spec = {
-    template,
-    css,
-    view: {
-        classes: ['press-excerpt']
-    },
-    model() {
-        return {
-            iconUrl: this.iconUrl,
-            author: this.author,
-            source: this.source,
-            org: this.org,
-            date: this.date,
-            headline: this.headline,
-            excerpt: this.excerpt,
-            url: this.url
-        };
-    }
-};
+function OptionalExcerpt({excerpt, url}) {
+    return (
+        excerpt &&
+            <div className="excerpt">
+                {excerpt}…{' '}
+                <a href={url}>Continue reading</a>
+            </div>
+    );
+}
 
-export default class PressExcerpt extends componentType(spec) {
+export default function PressExcerpt({iconUrl, author, date, source, url, headline, excerpt}) {
+    const ref = useRef();
 
-    onLoaded() {
-        if (this.iconUrl) {
-            this.el.classList.add('with-icon');
+    useEffect(() => {
+        if (iconUrl) {
+            ref.current.classList.add('with-icon');
```

Comment:
React is likely to not like you mucking with the DOM that it renders.  It seems like you could conditionally set `with-icon` in the class on the element when it renders instead.  the `classnames` lib makes that easy if you're using it

Reply:
Yes, this was a wrong decision for how to modify class list. But I thought part of the reason `useEffect` existed was to say "I'm done with the DOM, you can fiddle with it now."

Reply:
as a general rule you shouldn't change anything on the DOM that react has rendered directly.  You'd probably be ok modifying just the class though, but if you made more substantial changes it's likely that react would have errors if it re-rendered the content and it didn't match what it expected it to look like.

As I understand it, useEffect means react has rendered the content so you're able to *read* from it by measuring and kicking off animations.